### ISSUE TITLE
 Failed to install the following Android SDK packages as some licences have not been accepted.

### ISSUE THREAD

**Issue by ashLikun** [jitpack/jitpack.io#3687]
A problem occurred evaluating script.
> Failed to install the following Android SDK packages as some licences have not been accepted.
     platforms;android-27 Android SDK Platform 27
     build-tools;27.0.3 Android SDK Build-Tools 27.0.3
  To build this project, accept the SDK license agreements and install the missing components using the Android Studio SDK Manager.
  Alternatively, to transfer the license agreements from one workstation to another, see http://d.android.com/r/studio-ui/export-licenses.html
  
  Using Android SDK: /opt/android-sdk-linux

**Comment by zacharee** [comment]
Seeing as there are three other issues right before this one about the same problem, I'd say it's probably something up with the build servers.

https://github.com/jitpack/jitpack.io/issues/3686
https://github.com/jitpack/jitpack.io/issues/3685
https://github.com/jitpack/jitpack.io/issues/3684

**Comment by look0089** [comment]
I had the same problem.

**Comment by HamidrezaAmz** [comment]
I have same issue!:|
`* What went wrong:
A problem occurred configuring project ':app'.
> Failed to install the following Android SDK packages as some licences have not been accepted.
     build-tools;28.0.3 Android SDK Build-Tools 28.0.3
     platforms;android-28 Android SDK Platform 28
 `

**Comment by RX1226** [comment]
I had the same problem too!

**Comment by sicmu** [comment]
anyone resolve the problem?

**Comment by Hackforid** [comment]
The same with me

**Comment by avshvetsov** [comment]
Anyone resolve this issue?

**Comment by TonyTangAndroid** [comment]
@avshvetsov I have managed to resolve this by "hacking" around the script by adding command to accept the license. Please refer to this [commit](https://github.com/TonyTangAndroid/Secured-Preference-Store/commit/25c1b6ab31834168405059af7bd53fd1dd0d89c0) that have resolved my build.
This is the [log](https://jitpack.io/com/github/TonyTangAndroid/Secured-Preference-Store/0.5.4/build.log) that proves that it has built successfully.

Basically, you add one line of code at your project  file `graldew` with the following code 

`yes | $ANDROID_HOME/tools/bin/sdkmanager "build-tools;28.0.3"`

**Comment by giladna** [comment]
@avshvetsov 

Thanks!!!

  yes | $ANDROID_HOME/tools/bin/sdkmanager "platforms;android-28"

  yes | $ANDROID_HOME/tools/bin/sdkmanager "build-tools;27.0.3"

worked for me on gradlew

**Comment by noamtamim** [comment]
I have a safer solution. First of all, put this in jitpack.yml and not in gradlew -- so it will only affect JitPack and not others that happen to run your build.
Second, instead of explicitly installing the components, just tell Android you accepted the license.

## Step one
Copy this gist to some server you control (GitHub pages is a great option and so is a Gist):
https://gist.github.com/noamtamim/7ca339b68a03987171c04cd24caee1b3

## Step two
Create jitpack.yml (if you don't have one already) in the root of the repo:

```
jdk:
  - oraclejdk8 
before_install:
  - curl URL_OF_THE_FILE | sh
```

Example:

```
jdk:
  - oraclejdk8 
before_install:
  - curl https://gist.githubusercontent.com/noamtamim/7ca339b68a03987171c04cd24caee1b3/raw/15c5f4983a41d039b3a7f2cd980d54bbfc1412b2/android-sdk-license.sh | sh
```

Make sure the URL is to the RAW file.
But please copy this to your own account/gist/server. Every time there's an updated license, update this file with the new license hash. 
Legally if you run this script (or your own version of it), it means **you** accepted the license.
See here for more info: https://developer.android.com/studio/intro/update#download-with-gradle

**Comment by jitpack-io** [comment]
Hi all,

Should be fixed now. Thanks for the workarounds!

**Comment by tibbi** [comment]
seems to be fixed already, thanks :)

**Comment by avshvetsov** [comment]
@jitpack-io 
not all problems are resolved:
https://jitpack.io/com/github/avshvetsov/FastScroll/6ca6585d5a/build.log

Can you help?

**Comment by goui** [comment]
It is not working for me either
https://jitpack.io/com/github/goui/testlib/0.1.4/build.log

I have added the 2 lines at the end of my gradlew.
Thing is, even without those lines the ./gradlew install was working for me locally.

I think you might have a problem with your CI not accepting the licenses.

**Comment by SilenceDut** [comment]
@jitpack-io 
https://jitpack.io/com/github/silencedut/diffadapter/1.2.0/build.log
still not  be resolved in my project

**Comment by jitpack-io** [comment]
@SilenceDut Rebuilt v1.2.0

**Comment by oahmedazab** [comment]
Still not working for me :/
https://jitpack.io/com/github/inmobly/boltplay/1.0.18/build.log

**Comment by noamtamim** [comment]
> Still not working for me :/
> https://jitpack.io/com/github/inmobly/boltplay/1.0.18/build.log

@oahmedazab did you try my workaround? It's very easy to implement. See https://github.com/jitpack/jitpack.io/issues/3687#issuecomment-455901608.

**Comment by parcool** [comment]
Still not working for me ,but when i restart android studio it(the bug) disappear!

**Comment by jitpack-io** [comment]
Small clarification on how this works:
JitPack will try to read your build.gradle to find SDK and build-tools version. It will then install those versions. If you are using properties in your build.gradle, like this:
```
    compileSdkVersion rootProject.ext.android["compileSdkVersion"]
    buildToolsVersion rootProject.ext.android["buildToolsVersion"]
```
then JitPack is unable to detect those versions (yet).

For now we recommend using versions directly in build.gradle like this:
```
    compileSdkVersion "28"
    buildToolsVersion "28.0.3"
```

**Comment by Sangra69** [comment]
Nope the morons who decided to remove the --licenses option have no clue what they are doing. I am still getting that same error myself. It has to do with 28.0.3

I can't complete this bloody project now. Thanks android! 

Starting a Gradle Daemon, 2 incompatible and 1 stopped Daemons could not be reused, use --status for details

> Configure project :app
Checking the license for package Android SDK Build-Tools 28.0.3 in E:\android-sdk\licenses
Warning: License for package Android SDK Build-Tools 28.0.3 not accepted.
Checking the license for package Android SDK Platform 27 in E:\android-sdk\licenses
Warning: License for package Android SDK Platform 27 not accepted.

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':app'.
> Failed to install the following Android SDK packages as some licences have not been accepted.
     platforms;android-27 Android SDK Platform 27
     build-tools;28.0.3 Android SDK Build-Tools 28.0.3
  To build this project, accept the SDK license agreements and install the missing components using the Android Studio SDK Manager.
  Alternatively, to transfer the license agreements from one workstation to another, see http://d.android.com/r/studio-ui/export-licenses.html
  
  Using Android SDK: E:\android-sdk

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 11s

**Comment by noamtamim** [comment]
@Sangra69 Even though it seems to be working according to user feedback (I'm a JitPack user, not staff), have you tried the workaround I suggested here?
https://github.com/jitpack/jitpack.io/issues/3687#issuecomment-455901608

**Comment by huazaihuola** [comment]
Hi,I have the same problem, and I solved this problem by this way，my system is window10，and IDE is Android studio.

https://stackoverflow.com/questions/39760172/you-have-not-accepted-the-license-agreements-of-the-following-sdk-components/43003932#43003932?newreg=73869e982bcb4fa88fd8a1028649563d

**Comment by tarikguney** [comment]
> Still not working for me ,but when i restart android studio it(the bug) disappear!
This -- interestingly -- fixed my problem too...

**Comment by guplab** [comment]
Greetings!
Run the following command on the Terminal as administrator

`flutter doctor --android-licenses`

It should count the number of licenses that you not have accepted and then ask you for a review
press **Y** and then accept all the licenses you need.

I hope it helps, have a nice happy coding hours!
Cheers!!!

**Comment by bethuel11** [comment]
I downloaded build tools and SDK using Android Studio.

Hope this helps for anyone.

**Comment by mojimich2015** [comment]
> 
> 
> Small clarification on how this works:
> JitPack will try to read your build.gradle to find SDK and build-tools version. It will then install those versions. If you are using properties in your build.gradle, like this:
> 
> ```
>     compileSdkVersion rootProject.ext.android["compileSdkVersion"]
>     buildToolsVersion rootProject.ext.android["buildToolsVersion"]
> ```
> 
> then JitPack is unable to detect those versions (yet).
> 
> For now we recommend using versions directly in build.gradle like this:
> 
> ```
>     compileSdkVersion "28"
>     buildToolsVersion "28.0.3"
> ```

This Solved My Problem , Thank You

**Comment by Kravet** [comment]
Fixed when I changed the **bitbucket-pipelines.yml** file. I've added second licence value, because I had only one. You can check the values on your local machine in `$ANDROID_HOME/licenses` folder.

`- echo -e "\nd56f5187479451eabf01fb78af6dfcb131a6481e \n24333f8a63b6825ea9c5514f83c2829b004d1fee" > "$ANDROID_HOME/licenses/android-sdk-license"`

![image](https://user-images.githubusercontent.com/7354769/61694922-2790fd80-ad3b-11e9-8656-4949406205a6.png)

**Comment by outZoNe** [comment]
Just opened SDK Manager in Android Studio and install SDK version 28(Android Pie)

**Comment by samulon13** [comment]
locate tools\bin in your cmd then type sdkmanager --licenses.
if it says that all licenses are accepted but then youre still encountering the issue. 
try to restart your computer and rebuild again.

**Comment by usmanantharikta** [comment]
> locate tools\bin in your cmd then type sdkmanager --licenses.
> if it says that all licenses are accepted but then youre still encountering the issue.
> try to restart your computer and rebuild again.

$ANDROID_HOME/tools/bin/sdkmanager --licenses
that is work for me

**Comment by chrisMiller2** [comment]
For me it was just that my project was on an external hard drive and forgot to plug it in. So check it if thats the case!

**Comment by pavoltravnik** [comment]
> Hi,I have the same problem, and I solved this problem by this way，my system is window10，and IDE is Android studio.
> 
> https://stackoverflow.com/questions/39760172/you-have-not-accepted-the-license-agreements-of-the-following-sdk-components/43003932#43003932?newreg=73869e982bcb4fa88fd8a1028649563d

thank you, this actually works

**Comment by aveenashmandhwani** [comment]
Hey!!!

Just open terminal or git bash and head over to your android studio directory. 

For windows users: 
`cd C:\Users\{USERNAME}\AppData\Local\Android\Sdk\tools\bin`

and then run this to accept unaccepted licenses.
`sdkmanager.bat --licenses`

and then just review and accept them by typing 'y' and hit enter till its done..

Reference:

https://www.youtube.com/watch?v=c7HmwiTtYPE&t=580s

**Comment by pedrosqueiroz** [comment]
@aveenashmandhwani Worked like a charm! Thanks a lot for sharing. (on macOS just `open /usr/local/share/android-sdk` on your terminal)

**Comment by marciosarabando** [comment]
run command in cmd: **flutter doctor --android-licenses**

**Comment by ankitbhatia87** [comment]
![image](https://user-images.githubusercontent.com/7266667/67973298-45250b80-fc36-11e9-8aea-f7f0fe8366a8.png)

Just choose the selections as per the attached image in AndroidSDK >> SDK Tools it will work

**Comment by leandrocastro** [comment]
~/Library/Android/sdk/tools/bin
./sdkmanager --licenses

just this

**Comment by webismail92** [comment]
just install ur android studio update...

**Comment by al3ra8e628** [comment]
all what you need is to install them manually by the sdk-manager tool,

**Comment by jeffrey-coder** [comment]
anyone can assist me on this? i have no clue what sdkmanager is. im trying to call "cordova build android" and this error message popped. how do i solve this given that i have android studio already

**Comment by MichaelRSilva** [comment]
> > locate tools\bin in your cmd then type sdkmanager --licenses.
> > if it says that all licenses are accepted but then youre still encountering the issue.
> > try to restart your computer and rebuild again.
> 
> $ANDROID_HOME/tools/bin/sdkmanager --licenses
> that is work for me

That is work for me, Tks!

**Comment by DanFast** [comment]
> anyone can assist me on this? i have no clue what sdkmanager is. im trying to call "cordova build android" and this error message popped. how do i solve this given that i have android studio already

Install Android Studio, on Welcome to Android Studio Screen  on the right bottom you will see Configure, there choose SDK Manager ---> Appearance & Behavior --> System Settings --> Android SDK install Android version you gonna use maybe from 8.0 to latest..

This did the trick for me with Ionic 4 Cordova

**Comment by caiobraga98** [comment]
run the flutter console.Use the command "flutter doctor" and you gonna find the problem.This work for me.

**Comment by pimol** [comment]
https://github.com/jitpack/jitpack.io/issues/3684 closed for me.

**Comment by tonymedrano** [comment]
Android Studio: Appareance & Behavior/System Settings/Android sdk/SDK Update Sites
check checkbox Force https://... sources to be fetched using http://
don't forget to check any other third party you need.

**Comment by zoubir-remila** [comment]
You have just to install android sdk 27
TOOLS -> SDK MANAGER -> android 8.1  with api level 27 -> apply 
build a second time, It will works !!

**Comment by demolagiran** [comment]
In Android Studio 3.6 or later, you need to manually add the old version of the Android SDK Tools for Flutter to work. To do this:
Open the Android Studio SDK Manager
In the Android SDK tab, uncheck Hide Obsolete Packages
Check Android SDK Tools (Obsolete)
and install.

**Comment by NickelBalne** [comment]
> Greetings!
> Run the following command on the Terminal as administrator
> 
> `flutter doctor --android-licenses`
> 
> It should count the number of licenses that you have not accepted and then ask you for a review
> press **Y** and then accept all the licenses you need.
> 
> I hope it helps, have a nice happy coding hours!
> Cheers!!!

it worked me

**Comment by youguanxinqing** [comment]
> Greetings!
> Run the following command on the Terminal as administrator
> 
> `flutter doctor --android-licenses`
> 
> It should count the number of licenses that you have not accepted and then ask you for a review
> press **Y** and then accept all the licenses you need.
> 
> I hope it helps, have a nice happy coding hours!
> Cheers!!!

woked for me, thanks!

**Comment by XeroHero** [comment]
Similaar issue here when running builds on Travis CI for Android. 
"Failed to install the following Android SDK packages as some licences have not been accepted.

     build-tools;29.0.3 Android SDK Build-Tools 29.0.3"

**Comment by jefelon** [comment]
> ¡¡¡Oye!!!
> 
> Sólo tiene que abrir terminal o git bash y dirigirse a su directorio de estudio Android. Command Terminal se recomienda para usuarios de Windows.
> 
> Para usuarios de Windows:
> `cd C:\Users\{USERNAME}\AppData\Local\Android\Sdk\tools\bin`
> 
> y luego ejecutar esto para aceptar licencias no aceptadas.
> `sdkmanager.bat --licenses`
> 
> y luego simplemente revisar y aceptarlos escribiendo 'y' y pulse enter hasta que esté hecho..
> 
> Referencia:
> 
> https://www.youtube.com/watch?v=c7HmwiTtYPE&t=580s

Good

**Comment by alexjsmc** [comment]
Run   ubuntu 18.04 (SOLVED)

flutter doctor --android-licenses

**Comment by indatawetrust** [comment]
my solution
`android update sdk --no-ui --all --filter build-tools-28.0.3,android-28,extra-android-m2repository`

**Comment by rajawalid5d** [comment]
i have the same problem, and i switch my jdk into jdk8 following this guide https://dev.to/thegroo/install-and-manage-multiple-java-versions-on-linux-using-alternatives-5e93 then i run sdkmanager --licenses
then it run smooth
but make sure you have install jdk8 not only jre8
check your java version using java --version, android using java8

**Comment by Judiks** [comment]
try this
open cmd redirect to User\AppData\Local\Android\Sdk\tools\bin> then write **sdkmanager.bat --licenses**
User\AppData\Local\Android\Sdk\tools\bin>**sdkmanager.bat --licenses**

**Comment by kaxp** [comment]
I was getting this error when running my Flutter application :
Failed to install the following Android SDK packages as some licences have not been accepted. build-tools;28.0.3 Android SDK Build-Tools 28.0.3

Solution:
I tried sdkmanager.bat --licenses and flutter doctor --android-licenses, but it didn't help. 
So I manually downloaded the SDK build-tool 29.0.2 and put it inside Android/Sdk/build-tools folder.

**Comment by JakubGrzelak00** [comment]
I had the same problem. Try typing flutter doctor to cmd and then it will tell you what to do.

**Comment by samulon13** [comment]
@JakubGrzelak00  did you install android studio ?

**Comment by JakubGrzelak00** [comment]
@samulon13 Yes I did and face this error with first run on project. I'm new to flutter and this is the only working explanation that I found.

**Comment by samulon13** [comment]
this isnt about flutter. this is a common problem when we arent yet familiarize with android studio.

run sdkmanager --licenses in your terminal/cmd and accept licenses.
make sure you are in administrator mode.

**Comment by samulon13** [comment]
sometimes you might also need to reboot your system.

**Comment by rajanZoxima** [comment]
in build.gradle file ->dependencies->
    this was my class path ->   classpath("com.android.tools.build:gradle:3.4.2")
This is what i replaced it:
   classpath('com.android.tools.build:gradle:3.5.3')
     classpath 'com.google.gms:google-services:4.3.4'
upgrade gradle version it will work , this worked for me

**Comment by rajanrek** [comment]
in build.gradle file ->dependencies->
this was my class path -> classpath("com.android.tools.build:gradle:3.4.2")
This is what i replaced it:
classpath('com.android.tools.build:gradle:3.5.3')
classpath 'com.google.gms:google-services:4.3.4'
upgrade gradle version it will work , this worked for me

**Comment by athamour1** [comment]
I reinstall the build tools and this resolved th problem

**Comment by ncpy** [comment]
https://www.youtube.com/watch?v=g789PvvW4qo really helped me

**Comment by ebmbook** [comment]
Installing anything through SDK manager will automatically pops up a license agreement box and that's it. 

For example, install "Google Play Licensing Library":

`SDK Manager >> Android SDK >>  SDK Tools >> Google Play Licensing Library`

You will then be able to see the licenses folder under the  android installation folder for example:
  
`C:\Users\yourname\AppData\Local\Android\licenses`

**Comment by 12mohaned** [comment]
I believe the link below will help to resolve the issue 

https://vfac.fr/blog/fix-flutter-android-licenses-exception

**Comment by fengyongge** [comment]
$ANDROID_HOME/tools/bin/sdkmanager --licenses

That is work for me, Tks!

**Comment by vadim566** [comment]
just go reinstall the sdk, 
1.go to the sdk manager
2.check out the sdk u want to reinstall
3.after uninstall, recheck the sdk
4.and it should work
